File: android/app/src/main/java/com/dontworry/app/domain/thread/ThreadIdentity.kt
package com.dontworry.app.domain.thread

import com.dontworry.app.domain.text.TextNormalizer

object ThreadIdentity {
    fun fromParts(threadId: String?, threadLink: String?, threadTitle: String): String {
        val id = threadId?.trim().orEmpty()
        if (id.isNotEmpty()) return "id:$id"

        val link = threadLink?.trim().orEmpty()
        if (link.isNotEmpty()) return "link:$link"

        return "title:${TextNormalizer.normalize(threadTitle)}"
    }
}

File: ios/DontWorryiOS/Domain/ThreadIdentity.swift
import Foundation

enum ThreadIdentity {
    static func fromParts(threadId: String?, threadLink: String?, threadTitle: String) -> String {
        if let threadId = threadId?.trimmingCharacters(in: .whitespacesAndNewlines), !threadId.isEmpty {
            return "id:\(threadId)"
        }

        if let threadLink = threadLink?.trimmingCharacters(in: .whitespacesAndNewlines), !threadLink.isEmpty {
            return "link:\(threadLink.lowercased())"
        }

        return "title:\(TextNormalizer.normalize(threadTitle))"
    }
}
